src/akushon/action/process/joint_process.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021 Ichiro ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <string>
+#include <vector>
+
+#include "akushon/action/process/joint_process.hpp"
+
+#include "tachimawari/joint/model/joint.hpp"
+
+namespace akushon
+{
+
+JointProcess::JointProcess(uint8_t joint_id, float position)
+: joint(tachimawari::joint::Joint(joint_id, position)), initial_position(position),
+  target_position(position), additional_position(0.0)
+{
+}
+
+void JointProcess::set_target_position(float target_position, float speed)
+{
+  this->target_position = target_position;
+  additional_position = (this->target_position - initial_position) * speed;
```

Comment:
limit the minimum speed to 0 and maximum speed to 1

Reply:
done